data/model-db/generate_historical_data.R

```diff
@@ -0,0 +1,778 @@
+  mutate(run = "spring",
+         adipose_clipped = F) |> 
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) 
+
+try(if(any((unique(carcass_estimates$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(carcass_estimates$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(carcass_estimates$year)))
+  stop("Missing Year! Please fix!"))
+
+gcs_upload(carcass_estimates,
+           object_function = f,
+           type = "csv",
+           name = "model-db/carcass_estimates.csv",
+           predefinedAcl = "bucketLevel")
+
+# bulk carcass ------------------------------------------------------------
+# currently not going to use this table. originally designed to separate
+# out bulk counts from individual estimates
+
+# daily_redd --------------------------------------------------------------
+# TODO look into the unknow date entries for Feather River
+gcs_get_object(object_name = "standard-format-data/standard_daily_redd.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_daily_redd.csv",
+               overwrite = TRUE)
+daily_redd_raw <- read_csv("data/standard-format-data/standard_daily_redd.csv")
+
+daily_redd <- daily_redd_raw |> 
+  # remove the 100 or entries from feather river where date is unknown
+  # select for chinook 
+  filter(!is.na(date), species %in% c("chinook", "not recorded", "unknown")) |> 
+  select(date, latitude, longitude, reach, redd_id, age, age_index,
+         velocity, run, stream) |> 
+  left_join(survey_location, by = c("stream", "reach")) |>
+  select(-c(stream, reach, description)) |>
+  rename(survey_location_id = id) |>
+  # run_id
+  left_join(run, by = c("run" = "definition")) |> 
+  rename(run_id = id) |> 
+  select(-run, -description) 
+
+redd_reach <- daily_redd_raw |> 
+  group_by(stream) |> 
+  distinct(reach)
+
+try(if(any((unique(daily_redd$survey_location_id) %in% survey_location$id) == F)) 
+  stop("Missing Survey Location ID! Please fix!"))
+try(if(any((unique(daily_redd$run_id) %in% run$id) == F))
+  stop("Missing Run ID! Please fix!"))
+try(if(any(is.na(daily_redd$date)))
+  stop("Missing Date! Please fix!"))
+
+gcs_upload(daily_redd,
+           object_function = f,
+           type = "csv",
+           name = "model-db/daily_redd.csv",
+           predefinedAcl = "bucketLevel")
+
+# annual_redd -------------------------------------------------------------
+gcs_get_object(object_name = "standard-format-data/standard_annual_redd.csv",
+               bucket = gcs_get_global_bucket(),
+               saveToDisk = "data/standard-format-data/standard_annual_redd.csv",
+               overwrite = TRUE)
+annual_redd_raw <- read_csv("data/standard-format-data/standard_annual_redd.csv")
+
+annual_redd <- annual_redd_raw |> 
```

Comment:
we have an `annual_summarization_method` column that is not included in the database ERD. Confirming we want this column (seems good to have in there)

Reply:
I think this is needed - thank you for flagging. I will add to the ERD.

Reply:
Decided to remove the annual redd table and instead add mill to daily redd